internal/internal_task_handlers.go

```diff
@@ -618,13 +619,17 @@ func (wth *workflowTaskHandlerImpl) ProcessWorkflowTask(
 	if err != nil {
 		return nil, nil, err
 	}
-	defer workflowContext.Unlock()
+
+	defer func() {
+		workflowContext.Unlock(err)
+	}()
+
 
 	response, err := workflowContext.ProcessWorkflowTask(task, historyIterator)
 	return response, workflowContext, err
 }
 
-func (w *workflowExecutionContextImpl) ProcessWorkflowTask(task *s.PollForDecisionTaskResponse, historyIterator HistoryIterator) (completeRequest interface{}, err error) {
+func (w *workflowExecutionContextImpl)  ProcessWorkflowTask(task *s.PollForDecisionTaskResponse, historyIterator HistoryIterator) (completeRequest interface{}, err error) {
```

Comment:
Could you run `make fmt`? This is not formatted.

Reply:
sure. Will do